sdk-src/src/test/java/com/hpe/adm/nga/sdk/network/google/TestGoogleHttpClient.java

```diff
@@ -75,4 +93,95 @@ public void testRequestRetry() throws Exception {
                 times(GoogleHttpClient.getHttpRequestRetryCount() + 1))
                 .invoke("execute", any(), anyInt());
     }
+
+
+    @Test
+    public void testParallelRequestRetry() throws InterruptedException {
+        ConfigurationProperties.logLevel("WARN");
+
+        ClientAndServer clientAndServer = startClientAndServer();
+        Octane octane;
+        long totalExecutionTime = 500;
+        long cookieExpirationTime = 250;
+
+        setServerResponse(clientAndServer, cookieExpirationTime);
+        Authentication authentication = new SimpleUserAuthentication("", "");
+        String url = "http://localhost:" + clientAndServer.getLocalPort();
+        GoogleHttpClient spyGoogleHttpClient = spy(new GoogleHttpClient(url));
+        octane = new Octane.Builder(authentication, spyGoogleHttpClient).Server(url).workSpace(1002).sharedSpace(1001).build();
+
+        int nrCores = Runtime.getRuntime().availableProcessors();
+        Thread[] threads = new Thread[nrCores];
+        for (int i = 0; i < nrCores; i++) {
+            threads[i] = new Thread(() -> runGetRequest(octane, totalExecutionTime));
+            threads[i].start();
+        }
+        for (int i = 0; i < nrCores; i++) {
+            threads[i].join();
+        }
+        clientAndServer.stop();
+
+        //Exactly 2 authentications are done: one for 1st log in and 2nd for cookie refresh
+        Mockito.verify(spyGoogleHttpClient, times(2)).authenticate(any());
+
+        //There are more execution invocations than there are threads
+        Mockito.verify(spyGoogleHttpClient, atLeast(nrCores + 1)).execute(any());
+    }
+
+    public void runGetRequest(Octane octane, long totalExecutionTime) {
+        LocalDateTime pastTime = LocalDateTime.now();
+        LocalDateTime currentTime = LocalDateTime.now();
+        while (Duration.between(pastTime, currentTime).toMillis() < totalExecutionTime) {
+            octane.entityList("").get().execute();
+            currentTime = LocalDateTime.now();
+        }
+    }
+
+    public void setServerResponse(ClientAndServer clientAndServer, long cookieExpirationTime) {
```

Comment:
I would call this initServerResponse, the current naming suggest it's a setter for a field in the POJO